runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/KotlinGradlePluginVersion.kt

```diff
@@ -0,0 +1,30 @@
+package org.jetbrains.dokka.gradle.kotlin
+
+import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.plugin.getKotlinPluginVersion
+
+internal typealias KotlinGradlePluginVersion = KotlinVersion
+
+internal fun Project.getKgpVersion(): KotlinGradlePluginVersion? = parseKotlinVersion(this.getKotlinPluginVersion())
+
+/**
+ * Accepts a full version string that contains the major, minor
+ * and patch versions divided by dots, such as "1.7.10".
+ *
+ * Does NOT parse and store custom suffixes, so `1.8.20-RC2`
+ * or `1.8.20-dev-42` will be viewed as `1.8.20`.
+ */
+internal fun parseKotlinVersion(fullVersionString: String): KotlinVersion? {
+    val versionParts = fullVersionString
+        .split(".", "-", limit = 4)
```

Comment:
why is `4` rather than `3`?

Reply:
Limit sets the max number of substrings, so for 1.7.20-RC it'll be

[0] 1
[1] 7
[2] 20
[3] RC

If you use 3, then it'll be

[0] 1
[1] 7
[2] 20-RC